inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+            raise NotImplementedError(
+                "Remote execution is not supported for Segment Anything."
+            )
+        else:
+            raise ValueError(
+                f"Unknown step execution mode: {self._step_execution_mode}"
+            )
+
+    def run_locally(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+
+        predictions = []
+
+        if not boxes:
+            boxes = [None] * len(images)
+
+        for single_image, boxes_for_image in zip(images, boxes):
+            prompt_class_ids = []
+            prompt_class_names = []
+            prompt_index = 0
+
+            prompts = Sam2PromptSet()
+            if boxes_for_image is not None:
+                for x1, y1, x2, y2 in boxes_for_image.xyxy:
+
+                    prompt_class_ids.append(boxes_for_image.class_id[prompt_index])
+                    prompt_class_names.append(
+                        boxes_for_image.data["class_name"][prompt_index]
+                    )
+                    prompt_index += 1
+
+                    width = x2 - x1
+                    height = y2 - y1
+                    cx = x1 + width / 2
+                    cy = y1 + height / 2
+
+                    prompt = Sam2Prompt(
+                        box=Box(
+                            x=cx,
+                            y=cy,
+                            width=width,
+                            height=height,
+                        )
+                    )
+                    prompts.add_prompt(prompt)
+
+            inference_request = Sam2SegmentationRequest(
+                image=single_image.to_inference_format(numpy_preferred=True),
+                sam2_version_id=sam2_model,
+                api_key=self._api_key,
+                source="workflow-execution",
+                prompts=prompts,
+            )
+            sam_model_id = load_core_model(
+                model_manager=self._model_manager,
+                inference_request=inference_request,
+                core_model="sam2",
+            )
+
+            sam2_segmentation_response = self._model_manager.infer_from_request_sync(
+                sam_model_id, inference_request
+            )
+
+            prediction = self._convert_sam2_segmentation_response_to_inference_instances_seg_response(
+                sam2_segmentation_response.predictions,
+                single_image,
+                prompt_class_ids,
+                prompt_class_names,
+            )
+            predictions.append(prediction)
+
+        predictions = [
+            e.model_dump(by_alias=True, exclude_none=True) for e in predictions
+        ]
+        return self._post_process_result(
+            images=images,
+            predictions=predictions,
+        )
+
+    def _convert_sam2_segmentation_response_to_inference_instances_seg_response(
```

Comment:
make this an util function and test

Reply:
mhh, it seems similar to other utils function in that it is like conversion...but its very SAM / block result specific so don't think it will/should ever be used anywhere else.  but might be easier to test as non class functionI guess